src/lm/tools/summarizeTool.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import { InMemFileChange } from '../../common/file';
+import { FolderRepositoryManager } from '../../github/folderRepositoryManager';
+import { IssueModel } from '../../github/issueModel';
+import { PullRequestModel } from '../../github/pullRequestModel';
+import { RepositoriesManager } from '../../github/repositoriesManager';
+import { concatAsyncIterable } from './toolsUtils';
+
+interface SummarizationToolParameters {
+	issueNumber: number;
```

Comment:
The issue/pr should have already been fetched with the fetch tool. I would expect the parameter here to be the `FetchResult` (title, body, comments), not an issue number.